### PR TITLE
Feature/events v1

### PR DESCRIPTION
**Pull request by robYP**
新增 record table 功能 (create/update/delete event and saved in the record)

### CODE REVIEW

**Comment by robYP** [comment]
List events:
curl -X 'GET' \
  'http://127.0.0.1:8000/v1/events/?skip=0&limit=100' \
  -H 'accept: application/json'
  
 Create event:
  curl -X 'POST' \
  'http://127.0.0.1:8000/v1/events/' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0IiwiaWQiOjIsImV4cCI6MTcyNDM0NTg1Mn0.yE-CsLcLZAl11y_ArFWmYIfr3Q28RsCF0O0tqjlSBG8' \
  -H 'Content-Type: application/json' \
  -d '{
  "name": "ORDER_SHIPPED",
  "route": "LINE"
}'

Update event:
curl -X 'PUT' \
  'http://127.0.0.1:8000/v1/events/10' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0IiwiaWQiOjIsImV4cCI6MTcyNDM0NTg1Mn0.yE-CsLcLZAl11y_ArFWmYIfr3Q28RsCF0O0tqjlSBG8' \
  -H 'Content-Type: application/json' \
  -d '{
  "name": "ORDER_SHIPPED",
  "route": "TG"
}'

Delete event:
curl -X 'DELETE' \
  'http://127.0.0.1:8000/v1/events/10' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0IiwiaWQiOjIsImV4cCI6MTcyNDM0NTg1Mn0.yE-CsLcLZAl11y_ArFWmYIfr3Q28RsCF0O0tqjlSBG8'

File: api/v1/event.py [hunk 1]
```diff
@@ -19,7 +19,11 @@ def list_event(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
 
 @router.post("/")
 def create_event(user: Annotated[dict, Depends(get_current_user)], event: EventCreate, db: Session = Depends(get_db)):
-    event = repository.event.create_event(user, event, db)
+    event = repository.event.create_event(user=user, event=event, db=db)
```
Comment (jonyig) [inline api/v1/event.py]: 因應 service/repository pattern ，可以回想其定義，看是不是可以嘗試調整

File: devmentor-BE103-Python/versions/a652f9167dbe_add_on_delete_set_null_to_user_id_and_.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+down_revision: Union[str, None] = 'fdab6442de5f'
+branch_labels: Union[str, Sequence[str], None] = None
+depends_on: Union[str, Sequence[str], None] = None
+
+
+def upgrade() -> None:
+    op.drop_constraint('record_ibfk_1', 'record', type_='foreignkey')
+    op.drop_constraint('record_ibfk_2', 'record', type_='foreignkey')
+    
+    op.create_foreign_key(
```
Comment (jonyig) [inline devmentor-BE103-Python/versions/a652f9167dbe_add_on_delete_set_null_to_user_id_and_.py]: 經過上次討論，我們不使用 FK 顯性的機制，那這個 migration 還需要嗎？

Comment (robYP) [reply devmentor-BE103-Python/versions/a652f9167dbe_add_on_delete_set_null_to_user_id_and_.py]: 這個migration 不需要了
commit：
f51b778b2c27c358b16280c5ad1e75d8c7eac0a1

**Pull request merged**